filters/HeadFilter.cpp

```diff
@@ -50,4 +50,47 @@ std::string HeadFilter::getName() const
     return s_info.name;
 }
 
+void HeadFilter::addArgs(ProgramArgs& args)
+{
+    args.add("count", "Number of points to return from beginning.  "
+        "If 'invert' is true, number of points to drop from the beginning.",
+        m_count, point_count_t(10));
+    args.add("invert", "If true, 'count' specifies the number of points "
+        "to skip from the beginning.", m_invert, false);
+}
+
+bool HeadFilter::processOne(PointRef& point)
+{
+
+    bool keep = false;
+    if (m_index < m_count)
+        keep = true;
+    m_index++;
```

Comment:
m_index needs to be set to 0 in `ready()`

Reply:
it was set to 0 in the ctor. Is that not what we should be doing?

Reply:
This handles the case where someone may execute a stage more than once.